### DISCUSSION TITLE
Does Tailwind CSS support the new CSS :has selector?

### DISCUSSION THREAD

**Discussion by martinszeltins** [tailwindlabs/tailwindcss#10206] [Help] [answered]
There is a new CSS :has pseudo selector. I could not find anywhere if Tailwind already supports this or not.

**Comment by wongjn** [comment, marked as answer]
Yep! You can use it in your custom CSS, in variants from plugins, in arbitrary variants, etc.

**Reply by martinszeltins** [reply]
Can you provide a link to the tailwind documentation?

**Reply by wongjn** [reply]
https://tailwindcss.com/docs

**Reply by martinszeltins** [reply]
I meant the tailwind documentation where the :has selector is documented

**Reply by wongjn** [reply]
I'm not sure there are any specific mentions of the `:has()` selector in the Tailwind documentation.

**Reply by martinszeltins** [reply]
it does not seem to be working on https://play.tailwindcss.com/ and if it is supported then what is the syntax for it?

**Reply by wongjn** [reply]
Here some examples: https://play.tailwindcss.com/ruOlD4EYjJ

**Reply by martinszeltins** [reply]
> Here some examples: [play.tailwindcss.com/ruOlD4EYjJ](https://play.tailwindcss.com/ruOlD4EYjJ)

Thanks, that is what I was looking for.

````
<div class="[&:has(p)]:bg-red-500">
  <p></p>
</div>
````

**Reply by scheinercc** [reply]
@wongjn thanks for taking the time answering this. ~Do you know if there is anything more official than your example above?~ I guess there isn't much point for an official example [while Mozilla is still working on the feature](https://bugzilla.mozilla.org/show_bug.cgi?id=418039).

**Reply by dGilli** [reply]
The pseudo-class doesn’t seem to work in FF at the moment 👎  :(

There is an open bug for this: https://bugzilla.mozilla.org/show_bug.cgi?id=418039

**Reply by scheinercc** [reply]
> There is an open bug for this: https://bugzilla.mozilla.org/show_bug.cgi?id=418039

It is enabled in Nightly from v120, hopefully riding the train soon. 🤞🏻

**Comment by Apestein** [comment]
:has selector should be officially supported like group and peer. Styling base on parent and sibling is supported so it would only make sense to have support for styling base on child too.
https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-parent-state

**Reply by adamwathan** [reply]
Will definitely add native support for it soon, no resistance to it just 1000 things to do and can only do one at a time :)

**Reply by clarklab** [reply]
Definitely appreciate what must be a MOUNTAIN of a backlog, so I ask politely as possible: any other progress here? I'd love official :has support in Tailwind.

**Reply by martinszeltins** [reply]
I agree with @clarklab, the :has selector is becoming increasingly useful and would be awesome to have native support for it.

**Reply by Apestein** [reply]
This doesn't seem like it would be hard to support. Not sure what's the hold up.

**Comment by lexwebb** [comment]
In the meantime, you can also define a custom plugin in the tailwind .config file to use as a normal rule which tidies up the syntax a bit:

```javascript
plugins: [
  function ({ matchVariant }) {
    matchVariant(
      "has",
      (value) => {
        return `&:has(${value})`;
      },
      {
        values: {
          checked: "input:checked",
        },
      }
    );
  },
]
```

```html
<div class="has-checked:bg-red-500">
  <input type="radio" id={value} name={name} value={value} checked />
</div>
```


```html
<div class="has-[button:active]:bg-red-500">
  <button>I'm active</button>
</div>
```

**Comment by AlemTuzlak** [comment]
https://tailwindcss.com/blog/tailwindcss-v3-4#new-has-variant
it's here!

**Reply by shuvo-me** [reply]
Thanks for this share.
